File: src/discord/events/ready.ts
import { createEvent } from "#base";
import { ActivitiesOptions, ActivityType } from "discord.js";

createEvent({
    name: "ready",
    event: "ready",
    async run(client) {
        const status: ActivitiesOptions[] = [
            { name: "Powered by discloud", type:ActivityType.Custom },
            { name: "Join in: .gg/DBNATxA6Jx", type:ActivityType.Custom },
        ];

        let i = 0;
        setInterval(
            function() {
                client.user.setPresence({
                    status: "online",
                    activities: [status[i % status.length]]
                });
                i++;
            },
            15_000
        );        
    },
});

File: src/discord/events/ready.rs
use colored::Colorize;
use crate::discord::*;
use serenity::all::{ActivityData, Context, Ready};
use serenity::model::application::Command;

pub async fn run(app: &App, ctx: &Context, ready: &Ready) {
    log(&format!("● {} online ✓", ready.user.name.underline()).bright_green());
    let result = Command::set_global_commands(&ctx.http, &app.commands).await;
    let mut commands_len: usize = 0;
    match result {
        Ok(commands) => {
            commands_len = commands.len();
            log(&format!("└ {} command(s) successfully registered globally!",
                    commands_len
                ).bright_green(),
            );
            for command in commands {
                log(&format!("{{/}} Slash command > {} ✓",
                             command.name.bright_blue()).bright_green(),
                );
            }
        }
        Err(err) => {
            error(&format!("└ {} command(s) successfully registered globally!",
                    commands_len
                )
            );
            error(&format!("{:?}", err));
        }
    }

    ctx.set_activity(Some(ActivityData::custom(
        "Rust-powered bot.\nHosted by discloud.\nJoin in: .gg/DBNATxA6Jx",
    )));
}
